litestar/serialization.py

```diff
@@ -52,48 +44,96 @@
 T = TypeVar("T")
 
 
-def _enc_base_model(model: BaseModel) -> Any:
-    return model.dict()
+PYDANTIC_DECODERS: list[tuple[Callable[[Any], bool], Callable[[Any, Any], Any]]] = []
+
+try:
+    import pydantic
+
+    PYDANTIC_ENCODERS: dict[Any, Callable[[Any], Any]] = {
+        pydantic.EmailStr: str,
+        pydantic.NameEmail: str,
+        pydantic.ByteSize: lambda val: val.real,
+    }
+
+    def _dec_pydantic(type_: type[pydantic.BaseModel], value: Any) -> pydantic.BaseModel:
+        return type_.model_validate(value, strict=False) if hasattr(type_, "model_validate") else type_.parse_obj(value)
```

Comment:
Why not make this two separate functions depending on the Pydantic version?

Reply:
because this is unnecessary compilcation for an if else statement.